modules/web-console/frontend/app/components/page-queries/components/queries-notebook/controller.ts

```diff
@@ -2058,21 +2134,76 @@ export class NotebookCtrl {
             }
         };
 
-        $window.addEventListener('beforeunload', () => {
-            this._closeOpenedQueries(this.$scope.notebook.paragraphs);
+        const offTransitions = $transitions.onBefore({from: 'base.sql.notebook'}, ($transition$) => {
+            const options = $transition$.options();
+
+            if (options.custom.justIDUpdate || options.redirectedFrom)
```

Comment:
What's the point of this check? Looks like a copy/paste to me.

Reply:
Removed unused check.